docs/user-guide/query/gremlin/gremlin.md

```diff
@@ -270,3 +270,4 @@ for Gaffer specific options:
 | --- | --- | --- |
 | `operationOptions` | `g.with("operationOptions", "gaffer.federatedstore.operation.graphIds:graphA").V()` | Allows passing options to the underlying Gaffer Operations, this is the same as the `options` field on a standard JSON query. |
 | `getAllElementsLimit` | `g.with("getAllElementsLimit", 100).V()` | Limits the amount of elements returned if performing an unseeded query e.g. a `GetAllElements` operation. |
+| `hasStepFilterStage` | `g.with("hasStepFilterStage", "PRE_AGGREGATION").V()` | Controls which phase the filtering from a Gremlin `has()` stage is applied to the results. |
```

Comment:
Is this the right place to also put something about passing in username?

Reply:
I'm not sure we really want to advertise that as a feature as if using the `GafferPopAuthoriser` (which is essential for user authentication) it will block any attempts at setting the user id via a `with()` step.

Reply:
True, from a user perspective this probably isn't something that would be useful. It might be useful for an admin who is setting things up and testing them. Is this mentioned in the gremlin admin guide at all? It might be good to add into there, not something for this PR though.

Reply:
Hopefully its already covered in enough detail in the admin guide it mentions how setting the user ID will be denied etc https://gchq.github.io/gaffer-doc/latest/administration-guide/gaffer-deployment/gremlin.html#user-authentication